### DISCUSSION TITLE
Timestamp

### DISCUSSION THREAD

**Discussion by Pandagineer** [bitcraze/discussions#348] [Q&A] [answered]
I would like to know how long it takes for a portion of my C code to run on the Crazyflie. I found this possible solution:

```
#include <sys/time.h>

struct timeval stop, start;
gettimeofday(&start, NULL);
//do stuff
gettimeofday(&stop, NULL);
printf("took %lu us\n", (stop.tv_sec - start.tv_sec) * 1000000 + stop.tv_usec - start.tv_usec); 
```

(https://stackoverflow.com/questions/10192903/time-in-milliseconds-in-c)

But this doesn't work in the Crazyflie (I get an error when making). Should it work? If not, is there a solution for the Crazyflie? Thank you.

**Comment by knmcguire** [comment, marked as answer]
Hi! 

Have you tried the DEBUG_PRINT() function? That will print directly on the console of the cfclient.

I don't think printf is suitable for the crazyflie microprocessor

**Reply by Pandagineer** [reply]
This has been resolved. My goal was to collect data from the analog pins, along with a timestamp, and then write those data to a .csv file. Later I watched your YouTube video on how to add a variable to Log TOC, plot it, and save it as .csv (which includes the relative time). I successfully did this. Thank you!
